File: src/0012.js

var
  test = 500,
  n = 1,
  solution;

function divisors (n) {

  if (n === 1) return 1;

  var
    count = 2,
    i;

  for (i = 2; i <= n/2 + 1; i++) {
    if (n % i === 0) count++;
  }

  return count;
}

function triangle (n) {

  var
    sum = 0,
    i;

  for (i = n + 1; --i;) {
    sum += i;
  }

  return sum;
}

var d, t, boundary = Math.pow(10, 15);

while (++n) {
  t = triangle(n);
  d = divisors(t);
  if (d > 500) {
    solution = n;
    break;
  }
  if (t >= boundary) {
    console.log('fail!');
    break;
  }
}

console.log(t);

File: src/perl/0012.pl
#!/usr/bin/perl
my ($check,$sum);
$num = 1;
while (not $check) {
	$sum += $num;
	$check = &divisors($sum);
	print "$sum\n" if $check;
	$num ++;
}

sub divisors {
	my $count = 0;
	my $comp = shift;
	my $max = $comp;
	foreach my $x (1..($comp/2)) {
		last if $x > $max;
		$count += 2 and $max = $comp / $x if $comp % $x ==0;
	}
	return 1 if $count >= 500;
	return 0;
}